packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1840,6 +1889,45 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
 
     return errors;
   }
+
+  private validateInstanceMaintenancePolicy(policy: InstanceMaintenancePolicy | undefined) {
```

Comment:
I don't think it's necessary because passing the following existing validation will necessarily satisfy `maxHealthyPercentage > minHealthyPercentage`, but what do you think?

```ts
    if (
      maxHealthyPercentage !== undefined
      && maxHealthyPercentage !== -1
      && (maxHealthyPercentage < 100 || maxHealthyPercentage > 200)
    ) {
      throw new Error(`maxHealthyPercentage must be between 100 and 200, or -1 to clear the previously set value, got ${maxHealthyPercentage}`);
    }
    if (
      minHealthyPercentage !== undefined
      && minHealthyPercentage !== -1
      && (minHealthyPercentage < 0 || minHealthyPercentage > 100)
    ) {
      throw new Error(`minHealthyPercentage must be between 0 and 100, or -1 to clear the previously set value, got ${minHealthyPercentage}`);
    }
```